calypso/api_test.go

```diff
@@ -158,8 +160,48 @@ func TestClient_Calypso(t *testing.T) {
 	dk1, err := calypsoClient.DecryptKey(&DecryptKey{Read: *prRe1, Write: *prWr1})
 	require.Nil(t, err)
 	require.True(t, dk1.X.Equal(calypsoClient.ltsReply.X))
-	keyCopy1, err := DecodeKey(cothority.Suite, calypsoClient.ltsReply.X,
+	keyCopy1, err := decodeKey(cothority.Suite, calypsoClient.ltsReply.X,
 		dk1.C, dk1.XhatEnc, reader1.Ed25519.Secret)
 	require.Nil(t, err)
 	require.Equal(t, key1, keyCopy1)
+
+	// use keyCopy to unlock the stuff in writeInstance.Data
+}
+
+// decodeKey can be used by the reader of ByzCoin to convert the
+// re-encrypted secret back to a symmetric key that can be used later to decode
+// the document.
+//
+// Input:
+//   - suite - the cryptographic suite to use
+//   - X - the aggregate public key of the DKG
+//   - C - the encrypted key-slices
+//   - XhatEnc - the re-encrypted schnorr-commit
+//   - xc - the private key of the reader
+//
+// Output:
+//   - key - the re-assembled key
+//   - err - an eventual error when trying to recover the data from the points
+func decodeKey(suite kyber.Group, X kyber.Point, C kyber.Point, XhatEnc kyber.Point,
```

Comment:
is it better to be conservative and keep `DecodeKey` and `EncodeKey` for compatibility?

Reply:
It's in `_test`, so I'd keep it private.

Reply:
what I mean is the old `DecodeKey` and `EncodeKey` are public, they're in `struct.go`, so removing those will be, in theory, breaking compatibility

Reply:
According to our most recent definition of stability ("we are not willing to accept chagnes right now that destabilise current student projects based on cothority v3") then if these APIs are not useful, and not used, then removing them is ok. This was my reasoning for removing them; they seem to be only useful in test, and should never have been exported, and should not be in use now.